### PR TITLE
Properly configure changed optional dependencies 

### PR DESCRIPTION
**Pull request by elvece**
add logic to accept optional dependency list and determine dependencies based on new service configurations

### CODE REVIEW

File: system-images/compat/src/config/mod.rs [hunk 1]
```diff
@@ -1,26 +1,52 @@
...
     config_path: &Path,
+    maybe_deps_path: Option<&str>,
 ) -> Result<SetResult, anyhow::Error> {
     let rules: Vec<ConfigRuleEntry> = serde_yaml::from_reader(std::fs::File::open(rules_path)?)?;
     let mut cfgs = LinearMap::new();
     cfgs.insert(name, Cow::Borrowed(&config));
+
+    let mut depends_on = BTreeMap::new();
+    if maybe_deps_path.is_some() {
+        let deps_path = Path::new(maybe_deps_path.unwrap());
```
Comment (dr-bonez) [inline system-images/compat/src/config/mod.rs:38]:
```suggestion
    if let Some(deps_path) = maybe_deps_path.map(Path::new) {
```

Comment (elvece) [reply system-images/compat/src/config/mod.rs:38]: I can't grok this pattern for the life of me 😣 will update

Comment (ProofOfKeags) [reply system-images/compat/src/config/mod.rs:38]: matches also work if you forget the `if let` construct.

Comment (ProofOfKeags) [reply system-images/compat/src/config/mod.rs:38]: also:

```rust
if let Pat(...) = expr {
    do_x();
} else {
    do_y();
}
```
is *exactly* identical to
```rust
match expr {
    Pat(...) => { do_x(); }
    _ => { do_y(); }
}
```

**Review by dr-bonez — APPROVED** [review]

**Comment by ProofOfKeags** [comment]
squash this one.

**Pull request merged**